manifests/node_exporter.pp

```diff
@@ -112,6 +112,7 @@
   Optional[Array[String]] $collectors = undef,
   Array[String] $collectors_enable    = [],
   Array[String] $collectors_disable   = [],
+  String $scrape_job_name             = 'node',
```

Comment:
the problem is that traditionnally, `node` is used in job names everywhere in Prometheus, not `node_exporter`. and we don't have the actual `node` string anywhere in the module that we could use, so we do need to pass it along here somehow. we *could* hardcode it to `node` within the exporter class, but i figured it was a nice touch to allow overriding it because it *does* get changed sometimes... :/

Reply:
I'm not sure this prometheus naming tradition should play a role here. If we're managing prometheus using puppet, it's IMHO more useful to keep the cognitive load small by adhering to a single naming convention inside of the context where they are used (puppet), than adhering to a tradition from an outside context (prometheus).
After all, I suspect the average puppet-prometheus user will not be interested in the prometheus naming conventions. They will use `include ::prometheus::node_exporter` and will wonder why they then have to use `prometheus::scrape_jobs: - job_name: node (...)`.
But this just a gut feeling, of course.

Reply:
> After all, I suspect the average puppet-prometheus user will not be interested in the prometheus naming conventions.

Oh hell yeah they will. :) At least I am: the next step after setting up those exporters for me was to set up Grafana dashboards and *those* expected the job name to be `node`, not `node_exporter`. Rewriting all those dashboards would have been a massive pain in the bottom and it was much *MUCH* easier for me to patch the module this way. :)

Reply:
:thinking: hm, you might have a point.
Mismatch still feels wrong, IMHO, but since I can't suggest a better way to deal with it, I have to concede this one.